.github/workflows/tests-unit.yml

```diff
@@ -0,0 +1,136 @@
+name: Unit Tests
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - "**/*.rs"
+      - "**/Cargo.toml"
+      - "**/Cargo.lock"
+      - .config/nextest.toml
+      - .github/workflows/tests-unit.yml
+
+  push:
+    branches: [main]
+    paths:
+      - "**/*.rs"
+      - "**/Cargo.toml"
+      - "**/Cargo.lock"
+      - .config/nextest.toml
+      - .github/workflows/tests-unit.yml
+
+# Ensures that only one workflow task will run at a time. Previous builds, if
+# already in process, will get cancelled. Only the latest commit will be allowed
+# to run, cancelling any workflows in between
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  pull-requests: read
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+  RUST_LIB_BACKTRACE: 1
+  RUST_LOG: info
+  COLORBT_SHOW_HIDDEN: 1
+
+jobs:
+  unit-tests:
+    name: ${{ matrix.rust-version }} on ${{ matrix.os }}
+    permissions:
+      id-token: write
+      statuses: write
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 120
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        rust-version: [stable, beta]
+        features: [default-release-binaries]
+        exclude:
+          # Exclude macOS beta due to limited runner capacity and slower performance
+          - os: macos-latest
+            rust-version: beta
```

Comment:
I think we could also exclude windows+beta to save time

Reply:
Done [chore: remove extra windows run](https://github.com/ZcashFoundation/zebra/pull/9883/commits/fbcbd422f599fce584291b65b514b2b499dc382f)